### PR TITLE
Sharding extension for SIDB container image

### PR DESCRIPTION
**Pull request by abhisbyk**
Used **multi-stage builds**: 

- In first stage downloaded the entire repository from [oracle/db-sharding](https://github.com/oracle/db-sharding) using `Curl` utility. 
- In the second stage copy the necessary scripts from the previous stage, to the `$ORACLE_HOME/scripts/sharding` directory of the image to build sharding extension.

Signed-off-by: abhisbyk <abhishek.by.kumar@oracle.com>

### CODE REVIEW

**Comment by yunus-qureshi** [comment]
Please add a ReadMe

File: OracleDatabase/SingleInstance/extensions/sharding/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Sharding Extension
+Sharding extension is required to build the catalog and shard containers. When
```
Comment (cjbj) [inline OracleDatabase/SingleInstance/extensions/sharding/README.md:2]: -> "The sharding extension..."

File: OracleDatabase/SingleInstance/extensions/sharding/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+the SingleInstance container image is extended with **sharding** extension, it downloads the
```
Comment (cjbj) [inline OracleDatabase/SingleInstance/extensions/sharding/README.md:3]: -> "the **sharding** extension..."

File: OracleDatabase/SingleInstance/extensions/sharding/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Sharding Extension
+Sharding extension is required to build the catalog and shard containers. When
+the SingleInstance container image is extended with **sharding** extension, it downloads the 
+required scripts from [db-sharding/docker-based-sharding-deployment](https://github.com/oracle/db-sharding/tree/master/docker-based-sharding-deployment)
+repository, and package them with the SingleInstance container image to form extended image.
```
Comment (cjbj) [inline OracleDatabase/SingleInstance/extensions/sharding/README.md:5]: -> "packages"

-> "to form an extended image" (or "the extended")

File: OracleDatabase/SingleInstance/extensions/sharding/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+More information on catalog and shard containers can be found at `db-sharding/docker-based-sharding-deployment` [README](https://github.com/oracle/db-sharding/blob/master/docker-based-sharding-deployment/README.md).
```
Comment (cjbj) [inline OracleDatabase/SingleInstance/extensions/sharding/README.md:7]: That README itself is also missing a few "the"'s...

**Review by cjbj — CHANGES_REQUESTED** [review]

File: OracleDatabase/SingleInstance/extensions/sharding/README.md (hunk 1 above)
Comment (abhisbyk) [reply OracleDatabase/SingleInstance/extensions/sharding/README.md]: Done

File: OracleDatabase/SingleInstance/extensions/sharding/README.md (hunk 2 above)
Comment (abhisbyk) [reply OracleDatabase/SingleInstance/extensions/sharding/README.md]: Done

File: OracleDatabase/SingleInstance/extensions/sharding/README.md (hunk 3 above)
Comment (abhisbyk) [reply OracleDatabase/SingleInstance/extensions/sharding/README.md]: Done

File: OracleDatabase/SingleInstance/extensions/sharding/README.md (hunk 4 above)
Comment (abhisbyk) [reply OracleDatabase/SingleInstance/extensions/sharding/README.md:7]: @psaini79 Please take a look.

File: OracleDatabase/SingleInstance/extensions/sharding/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+# Run:
+#      $ docker build -t <extended_image_name> . --build-arg BASE_IMAGE=18.3.0-ee
+#
+
+ARG BASE_IMAGE_VERSION=19.3.0
+ARG BASE_IMAGE=oracle/database:${BASE_IMAGE_VERSION}-ee
+ARG BASE_REPO=https://github.com/oracle/db-sharding
+
+# Creating stage for downloading necessary script files
+FROM ${BASE_IMAGE} as downloader
```
Comment (yunus-qureshi) [inline OracleDatabase/SingleInstance/extensions/sharding/Dockerfile:25]: Can this first stage be from oracle-linux slim image

Comment (abhisbyk) [reply OracleDatabase/SingleInstance/extensions/sharding/Dockerfile:25]: I checked with oraclelinux:7-slim image, it does not have 'tar' or 'unzip' utilities. So, we can't use it in the first stage.

**Review by yunus-qureshi — APPROVED** [review]

File: OracleDatabase/SingleInstance/extensions/sharding/Dockerfile (hunk 5 above)
Comment (gvenzl) [reply OracleDatabase/SingleInstance/extensions/sharding/Dockerfile:25]: You could just install them via `yum` if that would make things easier/quicker.

Comment (abhisbyk) [reply OracleDatabase/SingleInstance/extensions/sharding/Dockerfile:25]: Thanks @gvenzl , now oraclelinux:7-slim is used in downloader stage. We only need this stage for downloading the archived repo and extract files, oraclelinux image will make things cleaner.

**Review by yunus-qureshi — APPROVED** [review]

File: OracleDatabase/SingleInstance/extensions/sharding/Dockerfile (hunk 5 above)
Comment (gvenzl) [reply OracleDatabase/SingleInstance/extensions/sharding/Dockerfile]: Yeah, I thought so. :)

File: OracleDatabase/SingleInstance/extensions/sharding/README.md (hunk 4 above)
Comment (psaini79) [reply OracleDatabase/SingleInstance/extensions/sharding/README.md:7]: Looks good to me

**Review by psaini79 — APPROVED** [review]

**Review by psaini79 — APPROVED** [review]

**Review by gvenzl — APPROVED** [review]

**Review by latakannan — APPROVED** [review]

**Review by gvenzl — APPROVED** [review]

**Pull request merged**